### DISCUSSION TITLE
Smarter compacting for the compacting upgrade?

### DISCUSSION THREAD

**Discussion by PokerfaceCactus** [P3pp3rF1y/SophisticatedBackpacks#504] [Ideas]
So I initially made the compacting upgrade thinking it would reduce the number of stacks in my backpack. but I quickly noticed that because it will try to compress as soon as possible I'll often end up with 2 stacks (uncompressed and compressed) very early on. 

would it be possible to have a mode for the compacting upgrade where it will only attempt to compact if either the compacted item already exists in the inventory or if more than 1 stack worth of a valid item is available in the inventory? that way it would only start compacting if it actually saved inventory space. 

for example with AllTheCompressed the compactor ends up making x1 compressed cobblestone the moment I get 9 cobblestone. 
and it'll make 2x compressed the second I have 9 of those. which means that at 91~ cobble stone total I'd have a regular stack, an x1 stack, and an x2 compressed stack. 
3 item stacks.

even though, if it had compacted only when the source item used more than 1 stack it would have resulted in the material only using 2 stacks in the end. making it more space-efficient.

**Comment by GFoniX** [comment]
I came here because I want to add a feature similar to this one! I thought about creating a post, but I checked in case it was a duplicate.

However, I have an issue on my end when I need to craft items and I require, for example, iron ingots. Often, I need 3 or 4 of them, but when I open my backpack, I see that everything is already compressed. So, I have to take a block back to decompress it every time I need it. It would be helpful, as mentioned in this comment, to have some kind of 'offset' so we can compress from a certain level (like 1 stack, 32, etc.), so we always have the item in question if we need to craft something with it.

**Reply by SU-5-KsavvaZ** [reply]
It will be great if compacting will occur only when slot is technically unable to store more items. Thats pretty easy to code and will solve both problems
